### DISCUSSION TITLE
Organization 2FA - GraphQL and REST api permission mismatch

### DISCUSSION THREAD

**Discussion by amitz-val** [community/community#45063] [Apps, API and Webhooks]
### Select Topic Area

Bug

## Expected Behavior
Application token with all permissions available should be able to get `organization.requiresTwoFactorAuthentication` via REST or Graphql

## Actual Behavior
The same token cannot get the value from the GraphQL but can get it in the REST api.

## Steps to reproduce
Graphql:
```
query {
  organization(login: "ORGNAME") {
    requiresTwoFactorAuthentication
  }
}
```

returns 
```
{
  "data": {
    "organization": {
      "requiresTwoFactorAuthentication": null
    }
  },
  "errors": [
    {
      "type": "FORBIDDEN",
      "path": [
        "organization",
        "requiresTwoFactorAuthentication"
      ],
      "extensions": {
        "saml_failure": false
      },
      "locations": [
        {
          "line": 3,
          "column": 5
        }
      ],
      "message": "Resource not accessible by integration"
    }
  ]
}
```

but via rest
```bash
curl \
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: Bearer SAMETOKEN"\
  -H "X-GitHub-Api-Version: 2022-11-28" \
  https://api.github.com/orgs/ORGNAME | jq
```

returns 
```
{
...
  "two_factor_requirement_enabled": true,
...
}
```

Same goes to `organization.membersWithRoles.edges.hasTwoFactorEnabled`.
Cannot get with any permission via graphql, but can get through rest api using `https://api.github.com/orgs/ORG/members?filter=2fa_disabled`

**Comment by amitz-val** [comment]
Bumpo

**Comment by rpgeddam** [comment]
I'm seeing a different message in the response that may be a bit more helpful:

>"message": "Your token has not been granted the required scopes to execute this query. The 'requiresTwoFactorAuthentication' field requires one of the following scopes: ['admin:org'], but your token has only been granted the: ['read:discussion', ... ] scopes. Please modify your token's scopes at: https://github.com/settings/tokens."

**Comment by faisalnazir7** [comment]


**Reply by tuves** [reply]
Hi @faisalnazir7, 

We’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. Thank you for helping us maintain an authentic and beneficial space for everyone.
